airbyte-integrations/connectors/source-recharge/source_recharge/components/recharge_error_handler.py

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from typing import Optional, Union
+
+from airbyte_cdk.sources.streams.http.error_handlers import HttpStatusErrorHandler
+from airbyte_cdk.sources.streams.http.error_handlers.response_models import ErrorResolution, FailureType, ResponseAction
+from requests import Response
+
+
+class RechargeErrorHandler(HttpStatusErrorHandler):
+    def __init__(self, logger: logging.Logger) -> None:
+        self.logger = logger
+        super().__init__(logger=logger)
+
+    def interpret_response(self, response_or_exception: Optional[Union[Response, Exception]] = None) -> ErrorResolution:
+
+        if isinstance(response_or_exception, Response):
+            content_length = int(response_or_exception.headers.get("Content-Length", 0))
```

Comment:
Seems like we need this custom handler, but can you add a comment about why we need the custom component. Basically that we need to inspect the header.

I think like its a feature gap that our http client isn't able to inspect the response headers and decide things on a predicate. The lines between what the declarative component vs the http client are starting to get a little blurry, but the more we see us having to implement a subclass of `HttpStatusErrorHandler`, the more it signals that it's lacking features.

Reply:
Yeah -- I've noticed a couple on-off checks in the existing `backoff_time` or `should_retry` methods. I feel like there hasn't been a consistent trend between these though.